src/cares_wrap.cc

```diff
@@ -122,10 +122,22 @@ struct node_ares_task {
   ChannelWrap* channel;
   ares_socket_t sock;
   uv_poll_t poll_watcher;
-  RB_ENTRY(node_ares_task) node;
 };
 
-RB_HEAD(node_ares_task_list, node_ares_task);
+struct TaskHash {
+  size_t operator()(node_ares_task* a) const {
+    return std::hash<node_ares_task*>()(a);
+  }
+};
+
+struct TaskEqual {
+  inline bool operator()(node_ares_task* a, node_ares_task* b) const {
+    return a->sock == b->sock;
+  }
+};
+
+typedef std::unordered_set<node_ares_task*, TaskHash, TaskEqual>
```

Comment:
> Maybe use `using ... = ...`, it's a little shorter.

Done. :)

> As well, you could replace TaskHash with a simple `using TaskHash = std::hash<node_ares_task*>;`, right?

That would make the hash set believe that any two different `node_ares_task*` pointers refer to different entries, but that’s not how we use it in the lookup below.